src/spec/controllers/functions_controller_spec.rb

```diff
@@ -21,6 +21,12 @@
       get :show, params: { id: function, user_id: user }
       expect(response).to be_successful
     end
+
+    it 'respond with the function file when the request url is ended with .rb' do
+      function = user.functions.create! valid_attributes
+      get :show, params: { id: function, user_id: user }, format: :rb
```

Comment:
ahaaa this is why when you were using `function_id: function` in `link_to` it was adding it as a parameter, because it should have been `id: function` :smile: